File: models/src/main/java/com/asadmshah/hnclone/errors/UsersServiceErrors.java
package com.asadmshah.hnclone.errors;

import io.grpc.Status;
import io.grpc.StatusRuntimeException;

public final class UsersServiceErrors {

    public static final Status USERNAME_EXISTS = Status
            .ALREADY_EXISTS
            .withDescription("Username exists.");

    public static final StatusRuntimeException USERNAME_EXISTS_EXCEPTION = USERNAME_EXISTS
            .asRuntimeException();

    public static final Status NOT_FOUND = Status
            .NOT_FOUND
            .withDescription("User not found.");

    public static final StatusRuntimeException NOT_FOUND_EXCEPTION = NOT_FOUND
            .asRuntimeException();

    public static final Status USERNAME_REQUIRED = Status
            .INVALID_ARGUMENT
            .withDescription("Username required.");

    public static final StatusRuntimeException USERNAME_REQUIRED_EXCEPTION = USERNAME_REQUIRED
            .asRuntimeException();

    public static final Status USERNAME_INVALID = Status
            .INVALID_ARGUMENT
            .withDescription("Username invalid.");

    public static final StatusRuntimeException USERNAME_INVALID_EXCEPTION = USERNAME_INVALID
            .asRuntimeException();

    public static final Status PASSWORD_INSECURE = Status
            .INVALID_ARGUMENT
            .withDescription("Insecure password.");

    public static final StatusRuntimeException PASSWORD_INSECURE_EXCEPTION = PASSWORD_INSECURE
            .asRuntimeException();

    public static final Status ABOUT_TOO_LONG = Status
            .INVALID_ARGUMENT
            .withDescription("Text entered for about is too long.");

    public static final StatusRuntimeException ABOUT_TOO_LONG_EXCEPTION = ABOUT_TOO_LONG
            .asRuntimeException();

    private UsersServiceErrors() {
    }
}

File: models/src/main/kotlin/com/asadmshah/hnclone/errors/UsersServiceErrors.kt
package com.asadmshah.hnclone.errors

import io.grpc.Status
import io.grpc.StatusRuntimeException

object UsersServiceErrors {

    val Exists: Status = Status
            .ALREADY_EXISTS
            .withDescription("Username exists.")

    val ExistsException: StatusRuntimeException = Exists
            .asRuntimeException()

    val NotFound: Status = Status
            .NOT_FOUND
            .withDescription("User not found.")

    val NotFoundException: StatusRuntimeException = NotFound
            .asRuntimeException()

    val UsernameRequired: Status = Status
            .INVALID_ARGUMENT
            .withDescription("Username required.")

    val UsernameRequiredException: StatusRuntimeException = UsernameRequired
            .asRuntimeException()

    val UsernameInvalid: Status = Status
            .INVALID_ARGUMENT
            .withDescription("Unable to use this username.")

    val UsernameInvalidException: StatusRuntimeException = UsernameInvalid
            .asRuntimeException()

    val PasswordInsecure: Status = Status
            .INVALID_ARGUMENT
            .withDescription("Please use a more secure password.")

    val PasswordInsecureException: StatusRuntimeException = PasswordInsecure
            .asRuntimeException()

    val AboutTooLong: Status = Status
            .INVALID_ARGUMENT
            .withDescription("Unable to use text for about.")

    val AboutTooLongException: StatusRuntimeException = AboutTooLong
            .asRuntimeException()
}